### PR TITLE
[AMDGPU][True16][MC] add true16 and fake16 test file for vop3 instructions

### PR DESCRIPTION
**Pull request by broxigarchen**
duplicating mc test, and updating proper flag for true16 and fake16 test file for vop3 instructions. This is preparing for the up-coming VOP3 true16 changes

### CODE REVIEW

File: llvm/test/MC/AMDGPU/gfx12_asm_vop3_dpp16-fake16.s [hunk 1]
```diff
@@ -0,0 +1,5764 @@
+// RUN: not llvm-mc -triple=amdgcn -mcpu=gfx1200 -mattr=+wavefrontsize32,-real-true16 -show-encoding %s | FileCheck --check-prefixes=GFX12,W32 %s
+// RUN: not llvm-mc -triple=amdgcn -mcpu=gfx1200 -mattr=+wavefrontsize64,-real-true16 -show-encoding %s | FileCheck --check-prefixes=GFX12,W64 %s
+// RUN: not llvm-mc -triple=amdgcn -mcpu=gfx1200 -mattr=+wavefrontsize32,-real-true16 %s 2>&1 | FileCheck --check-prefixes=GFX12-ERR,W32-ERR --implicit-check-not=error: %s
+// RUN: not llvm-mc -triple=amdgcn -mcpu=gfx1200 -mattr=+wavefrontsize64,-real-true16 %s 2>&1 | FileCheck --check-prefixes=GFX12-ERR,W64-ERR --implicit-check-not=error: %s
+
+v_add3_u32_e64_dpp v5, v1, v2, v3 quad_perm:[3,2,1,0]
```
Comment (kosarev) [inline llvm/test/MC/AMDGPU/gfx12_asm_vop3_dpp16-fake16.s:6]: *-fake16 tests are normally supposed to only contain True16 instructions?

Comment (Sisyph) [reply llvm/test/MC/AMDGPU/gfx12_asm_vop3_dpp16-fake16.s:6]: I think it does provide some additional code coverage to test the non-16-bit instructions in both real16 and fake16 modes. We can fully test both paths. But it is a small amount of additional coverage.

Comment (kosarev) [reply llvm/test/MC/AMDGPU/gfx12_asm_vop3_dpp16-fake16.s:6]: Keeping the non-True16 instructions would double the work updating them, and I'm not sure how much value there is in testing that they are not affected by the real-true16 attribute. I think I'd prefer to not have them, but don't insist.

Comment (rampitec) [reply llvm/test/MC/AMDGPU/gfx12_asm_vop3_dpp16-fake16.s:6]: It exists in the gfx12_asm_vop3_dpp16.s already. IMHO it is better to remove it from t16 tests.

**Comment by broxigarchen** [comment]
It seems more people want to remove the non-16-bit from the test file. Since there are more files with this same problem, I will first push this and open another PR to remove them altogether

**Pull request merged**